### PR TITLE
Include functions from the deprecated cgi module

### PR DESCRIPTION
**Pull request by trendels**
The stdlib `cgi` module is deprecated and going to be removed in Python 3.13. Currently, if you run `pytest` on any file that uses `mimeparse` you will see this deprecation warning:

~~~
========================================== warnings summary ===========================================
mimeparse.py:1
  /Users/st/p/python-mimeparse/mimeparse.py:1: DeprecationWarning: 'cgi' is deprecated and slated for removal in Python 3.13
    import cgi

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
~~~

Since mimeparse only depends on two (short) functions from the `cgi` module, I suggest to include them directly in the mimparse codebase. This will silence the warning and make sure mimeparse will continue to work with Python >= 3.13.

Edit: Source: https://github.com/python/cpython/blob/3.11/Lib/cgi.py#L226-L256

### CODE REVIEW

File: mimeparse.py
Comment (CaselIT) [inline mimeparse.py:25]: let's indicate from where _parse_header and _parseparam are taken.

Something like "Vendored version of cgi.parse_header from python 3.12" or something

**Review by CaselIT — COMMENTED** [review]
looks ok, just left a comment

File: mimeparse.py
Comment (trendels) [reply mimeparse.py:29]: I've added comments.

**Review by CaselIT — APPROVED** [review]
looks ok to me. I'll wait to hear from of the other maintainers before merging

**Comment by rafalkrupinski** [comment]
@CaselIT no one makes any complaints; any chance of merging it?

**Comment by vytas7** [comment]
Hi @rafalkrupinski, yes, we will address this before the new release. We'll likely want to synchronize this with the Falcon framework. There are different options going forward, as Falcon uses only a subset of this library (which is already very small), so we might want to unvendor and reimplement that completely, and backport a more performant version from Falcon. Which does not exist yet in the main tree, but I'm working on it in my branch.

**Comment by vytas7** [comment]
We can merge this as-is for now though, and improve later on. Thanks for this PR!

**Pull request merged**
